setup.py

```diff
@@ -126,6 +126,9 @@
         'invenio_base.apps': [
             'invenio_oauthclient = invenio_oauthclient:InvenioOAuthClient',
         ],
+        'invenio_base.api_apps': [
```

Comment:
As far as I know, it's an issue related only with [cern authentication plugin](https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/contrib/cern.py#L329).
The plugin is listening for user login to enrich the user informations (in this case with the list of groups).
So, if you register through the CERN plugin and then try to access with a [oauth2server](https://github.com/inveniosoftware/invenio-oauth2server/blob/master/invenio_oauth2server/views/server.py#L57) token, invenio will fail because the application `oauthclient` is not registered.

```python
Traceback (most recent call last):
  File "/opt/cds/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/opt/cds/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 181, in handle_request
    for item in respiter:
  File "/opt/cds/lib/python2.7/site-packages/raven/middleware.py", line 35, in __call__
    iterable = self.application(environ, start_response)
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/opt/cds/lib/python2.7/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/opt/cds/lib/python2.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/cds/lib/python2.7/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/opt/cds/lib/python2.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1610, in full_dispatch_request
    rv = self.preprocess_request()
  File "/opt/cds/lib/python2.7/site-packages/flask/app.py", line 1831, in preprocess_request
    rv = func()
  File "/opt/cds/src/invenio-oauth2server/invenio_oauth2server/ext.py", line 174, in verify_oauth_token_and_set_current_user
    valid, req = func(valid, req)
  File "/opt/cds/src/invenio-oauth2server/invenio_oauth2server/views/server.py", line 58, in login_oauth2_user
    identity=Identity(oauth.user.id))
  File "/opt/cds/lib/python2.7/site-packages/blinker/base.py", line 267, in send
    for receiver in self.receivers_for(sender)]
  File "/opt/cds/src/invenio-oauthclient/invenio_oauthclient/contrib/cern.py", line 344, in on_identity_changed
    remote = find_remote_by_client_id(client_id)
  File "/opt/cds/src/invenio-oauthclient/invenio_oauthclient/contrib/cern.py", line 180, in find_remote_by_client_id
    for remote in current_oauthclient.oauth.remote_apps.values():
  File "/opt/cds/lib/python2.7/site-packages/werkzeug/local.py", line 343, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/opt/cds/lib/python2.7/site-packages/werkzeug/local.py", line 302, in _get_current_object
    return self.__local()
  File "/opt/cds/src/invenio-oauthclient/invenio_oauthclient/proxies.py", line 31, in <lambda>
    lambda: current_app.extensions['invenio-oauthclient']
KeyError: 'invenio-oauthclient'
```